File: src/medium/container-with-most-water/1.go
package main

import (
	"fmt"
	"math"
)

/*
 * @lc app=leetcode id=11 lang=golang
 *
 * [11] Container With Most Water
 */

func main() {
	ary := []int{2, 3, 4, 5, 18, 17, 6}
	// ary := []int{1, 8, 6, 2, 5, 4, 8, 3, 7}
	fmt.Println(maxArea(ary))
}

// @lc code=start
func maxArea(height []int) int {
	result := 0
	for i, j := 0, len(height)-1; i < j; {
		curr := math.Min(float64(height[i]), float64(height[j])) * float64(j-i)
		if float64(result) < curr {
			result = int(curr)
		}
		if height[i] < height[j] {
			i++
		} else {
			j--
		}
	}

	return result
}

// @lc code=end

File: src/easy/climb-stairs/1.ts
/*
 * @lc app=leetcode id=70 lang=typescript
 *
 * [70] Climbing Stairs
 */

// @lc code=start
function climbStairs(n: number): number {
  if (n == 1) {
    return 1;
  }
  if (n == 2) {
    return 2;
  }
  let dp = new Array(n).fill(-1);
  dp[0] = 1;
  dp[1] = 2;
  for (let i = 2; i < n; i++) {
    dp[i] = dp[i - 1] + dp[i - 2];
  }
  return dp[n - 1];
}

// @lc code=end
